lib/userMemoryStore-spec.js

```diff
@@ -0,0 +1,260 @@
+        assert(!err);
+        assert(daisy.username === 'daisy');
+        done();
+      });
+    });
+    it('should not allow edits', function(done) {
+      store.read(daisyId, function(err, daisy) {
+        assert(!err);
+        daisy.username = 'donald duck';
+        store.read(daisyId, function(err, daisy2) {
+          assert(!err);
+          assert.equal(daisy2.username, 'daisy',
+            'username should not have been edited');
+          done();
+        });
+      });
+    });
+    it('should error when not found', function(done) {
+      store.read('invalid_id', function(err, user) {
+        assert(err);
+        assert(!user);
+        done();
+      });
+    });
+  });
+  describe('#byUsername', function() {
+    it('should find an user by username', function(done) {
+      store.byUsername('daisy', function(err, daisy) {
+        assert(!err);
+        assert(daisy.username === 'daisy');
+        done();
+      });
+    });
+    it('should not allow edits', function(done) {
+      store.byUsername('daisy', function(err, daisy) {
+        assert(!err);
+        daisy.username = 'donald duck';
+        store.byUsername('daisy', function(err, daisy2) {
+          assert(!err);
+          assert(daisy2);
+          done();
+        });
+      });
+    });
+    it('should error when not found', function(done) {
+      store.byUsername('invalid_username', function(err, user) {
+        assert(err);
+        assert(!user);
+        done();
+      });
+    });
+  });
+  describe('#create', function() {
+    it('should add a new user [slow]', function(done) {
+      var oldCount;
+      store.all(function(err, orig) {
+        assert(!err);
+        oldCount = orig.length;
+        store.create(userToCreate, function(err) {
+          assert(!err),
+          store.all(function(err, newUsers) {
+            assert(!err);
+            assert.equal(newUsers.length, oldCount + 1,
+              'total users should have increased by 1');
+            done();
+          });
+        });
+      });
+    });
+    it('should generate an id [slow]', function(done) {
+      store.create(userToCreate, function(err, user) {
+        assert(!err);
+        assert(user.id);
+        done();
+      });
+    });
+  });
+  describe('#update', function() {
+    it('should update fields', function(done) {
+      store.update(daisyId, {position: 'test'}, function(err, newDaisy) {
+        assert(!err);
+        assert.equal(newDaisy.username, 'daisy');
+        assert.equal(newDaisy.position, 'test');
+        done();
+      });
+    });
+    it('should error when not found', function(done) {
+      store.update('invalid_id', {position:'test'}, function(err, user) {
+        assert(err);
+        assert(!user);
+        done();
+      });
+    });
+  });
+  describe('#verifyPassword', function() {
+    beforeEach(function(done) {
+      var self = this;
+      store.create(userToCreate, function(err, user) {
+        self.user = user;
```

Comment:
I'm not actually using it again, but mocha allows you to add stuff to `this` in `beforeEach` hooks so it becomes available to the test functions